internal/storage/elasticsearch/config/config.go

```diff
@@ -501,6 +518,26 @@ func (c *Configuration) getConfigOptions(ctx context.Context, logger *zap.Logger
 	}
 	options = append(options, elastic.SetHttpClient(httpClient))
 
+	// API Key authentication
+	if c.Authentication.APIKeyAuthentication.FilePath != "" {
+		apiKeyFromFile, err := loadTokenFromFile(c.Authentication.APIKeyAuthentication.FilePath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to load API key from file: %w", err)
+		}
+		// Store loaded API key in context for use by GetHTTPRoundTripper
+		ctx = apikey.ContextWithAPIKey(ctx, apiKeyFromFile)
```

Comment:
I don't follow. What does this have to do with config / initialization? Tokens are only used at runtime when serving requests.